src/client/ServerMessageFactory.ts

```diff
@@ -0,0 +1,121 @@
+// The Ed-Fi Alliance licenses this file to you under the Ed-Fi License Agreement.
+// See the LICENSE file in the project root for more information.
+
+import path from 'node:path';
+import type { OutputChannel } from 'vscode';
+import { findDataStandardVersions, MetaEdConfiguration, newMetaEdConfiguration, SemVer } from '@edfi/metaed-core';
+import { findMetaEdProjects } from './ProjectFinder';
+import { getTargetOdsApiVersionSemver, allianceMode, getOdsApiDeploymentDirectory } from './ExtensionSettings';
+import { WorkspaceProjects } from '../model/WorkspaceProjects';
+import { ServerMessage } from '../model/ServerMessage';
+import { showErrorNotification, showInfoNotification } from './Utility';
+import {
+  bundledDsRootPath,
+  odsApiVersionSupportsDsVersion,
+  dsVersionToModelProjectDirectory,
+  odsApiVersionSupports,
+} from './DataStandardManager';
+
+type ServerMessageFactoryOptions = {
+  showUiNotifications: boolean;
+};
+
+/**
+ * Log error message to console and UI (if UI notifications are requested)
+ */
+async function notifyError(errorMessage: string, outputChannel: OutputChannel, showUiNotifications: boolean) {
+  if (showUiNotifications) {
+    await showErrorNotification(errorMessage);
+  }
+  outputChannel.appendLine(errorMessage);
+}
+
+/**
+ * Log info message to console and UI (if UI notifications are requested)
+ */
+async function notifyInfo(infoMessage: string, outputChannel: OutputChannel, showUiNotifications: boolean) {
+  if (showUiNotifications) {
+    await showInfoNotification(infoMessage);
+  }
+  outputChannel.appendLine(infoMessage);
+}
+
+/**
+ * Creates a ServerMessage for lint/build/deploy from the VS Code workspace and settings
+ */
+export async function createServerMessage(
+  outputChannel: OutputChannel,
+  { showUiNotifications }: ServerMessageFactoryOptions = { showUiNotifications: true },
+): Promise<ServerMessage | undefined> {
+  const { projectMetadatas, invalidProjects }: WorkspaceProjects = await findMetaEdProjects();
+
+  if (projectMetadatas.length === 0) {
+    // There are no MetaEd projects in the workspace
+    await notifyError(
+      `MetaEd requires a Data Standard project in the workspace. Bundled versions are at ${bundledDsRootPath()}`,
+      outputChannel,
+      showUiNotifications,
+    );
+    return undefined;
+  }
+
+  if (invalidProjects.length !== 0) {
+    // There are non-MetaEd projects in the workspace
+    await notifyInfo(
+      'There are non-MetaEd projects in the workspace. They will be ignored',
```

Comment:
I'm getting this message on my extension folder, and it is preventing me from running any actions on the extension.
![image](https://user-images.githubusercontent.com/9324390/220417501-e891a965-eb44-498b-9715-1d5a7631182e.png)

Reply:
Capital N?

Reply:
The blue squiggle was from the spellchecker, and I ignored that. But you're right, that was the problem. Things are working much better now :-).

Reply:
This is a good one for me to include in the troubleshooting guide.